crates/matrix-sdk-base/src/client.rs

```diff
@@ -1098,27 +1096,35 @@ impl BaseClient {
     /// Receive a get member events response and convert it to a deserialized
     /// `MembersResponse`
     ///
+    /// This client-server request must be made without filters to make sure all members are received. Otherwise, an error is returned.
     ///
     /// # Arguments
     ///
     /// * `room_id` - The room id this response belongs to.
     ///
     /// * `response` - The raw response that was received from the server.
     #[instrument(skip_all, fields(?room_id))]
-    pub async fn receive_members(
+    pub async fn receive_all_members(
         &self,
         room_id: &RoomId,
+        request: &api::membership::get_member_events::v3::Request,
         response: &api::membership::get_member_events::v3::Response,
-    ) -> Result<MembersResponse> {
+    ) -> Result<()> {
+        if request.membership.is_some() || request.not_membership.is_some() || request.at.is_some()
+        {
+            return Err(Error::ApiMisuse);
+        }
```

Comment:
Let's also add a comment why we have this restriction, and maybe link to this PR so future us can look at the previous state, if we wanted to add the new public APIs later.